### DISCUSSION TITLE
Estou iniciando e estou perdido.

### DISCUSSION THREAD

**Discussion by ghost** [frontendbr/forum#717] [Discussão]
Bem, sou novo na comunidade e estou iniciando os estudos juntando com a faculdade de ADS. Porém, atualmente eu ando muito perdido, vou contar o que está acontecendo. 
Estou no primeiro semestre e não estou conseguindo organizar os estudos corretamente, sendo que, até agora  só fiquei no "vou fazer" e simplesmente não consigo separar um tempo para isso e acabo procrastinando de uma maneira grande. Quero iniciar os estudos em Front End para assim, conseguir um estágio futuramente. Eu já vi várias técnicas de estudo, tentei utilizar a técnica pomodoro quando eu estava estudando lógica a parte, antes de iniciar a faculdade mas não me senti muito produtivo e acabei tendo um burnout e acho que está prestes a acontecer o mesmo até o final desse mês. As crises de ansiedade estão forte e não estou conseguindo me organizar. Fico procurando maneiras de me organizar, faço listas, já fiz várias listas mas não consigo sair da procrastinação. Como vocês lidam com isso? 
E bem, quero começar a estudar HTML/CSS/JavaScript em breve. Porém, algo que me empaca é a prática. Consigo entender tranquilamente, mas quando é pra desenvolver algo próprio, reproduzir, ou criar algum projeto, fico muito perdido e isso me faz procrastinar mais. Entendo a lógica passada durante a aula sem problemas, mas para emplementar no código tenho uma certa dificuldade. É normal isso? E quais técnicas de estudo que vocês utilizam e como se organizam para isso? Pois atualmente, tudo o que eu quero fazer é chegar da faculdade e dormir. Estou vendo que estou acumulando matéria, tenho um bom tempo livre na faculdade, mas não consigo utiliza-lo para estudo e sim para procrastinação e quero mudar esse hábito o mais rápido possível. 
Desculpem por essa introdução estranha no fórum mas isso é algo que tem me encomodado bastante. Já tentei várias alternativas e parece que procrastinar é a única que funciona haha 
Desde já, agradeço qualquer ajuda. (:  
1

**Comment by Pliavi** [comment]
Hehe, também entrei aqui esses dias, to na mesma que tu, a diferença é que estou no 6º semestre :smile:
Como lidar com a procrastinação, também não sei, mas já estou há 6 anos (desdo técnico em web) levando e até que está bem, talvez seja normal, ora ou outra tu pode tentar se controlar, com algo tipo StayFocused quando realmente precisa fazer algo, uma garrafinha de água na mesa também ajuda, mas nada que faça milagres.
Ás vezes você para de mexer pela frustração, pra quem está começando é normal, você mesmo disse, entende, mas não consegue fazer, isso pode ter 2 motivos:
- Ainda não tem prática, parece que tudo é processo criativo, mas tem mutia coisa mecânica, com o tempo fica mais fácil. Ex: Eu mesmo demorei pra entender as arrays (principalmente por a maioria iniciar em 0), algum tempo depois isso fica tão fixo na tua cabeça que se pegar uma que inicia em um vai achar estranho.
- Você deve estar pensando alto demais, tente pegar coisas simples, nem que seja criar um botão bonitinho com CSS, ou fazê-lo mostrar uma mensagem escondida com JS e vá aumentando aos poucos.

Ah, uma dica, papel e caneta ajuda bastante, não pra escrever código no papel (há quem diga que ajuda a decorar, mas a diferença caneta-teclado acho que atrapalha), mas desenhe o que tu quer fazer, tente rabiscar como as coisas funcionam, mais pra frente você verá teste de mesa, é chato, mas ajuda a entender porque algo não está funcionando.

Também não tenho muita experiência, mas espero ter ajudado com algo. Siga na luta man!

**Comment by Vitor-Felipe** [comment]
@brunojokubauskas Já passei por está situação também kkkk
Uma solução que eu encontrei para a procrastinação, foi o [trello](https://trello.com/), com ele eu consigo planejar toda a minha semana.
E em relação a prática, acho que seria bom você fazer alguns cursos online, como:
[Curso em video](http://www.cursoemvideo.com/course/curso-de-html5/),
[Caelum](https://www.caelum.com.br/apostila-html-css-javascript/),
[w3schools](https://www.w3schools.com/html/default.asp),
[Alura](https://www.alura.com.br/carreira-desenvolvedor-front-end) (esse é pago, mas é muito bom!!!)

Espero ter ajudado e boa sorte nos estudos!!

**Comment by ghost** [comment]
Eu assino o serviço da Alura. Já tentei o Trello, mas acredita que não consigo sair da procrastinação? Da ultima vez que tive o burnout, eu estava estudando praticamente 24h por dia rs pois tinha bastante tempo livre. O Trello já baixei, tentei utilizar, mas não consigo criar o maldito hábito de olhar o aplicativo.  Tento colocar listas em papel pra ver se sai, plano de estudos, mas não vai. Estou prestes a me estressar pois essa semana começa as provas do mês, não to conseguindo me organizar pra fazer as listas de exercício, estudar matéria que não entendi e entendi em partes, rs. To empacado e entrando em desespero. Fora os trabalhos, seminário e projetos que já tem. Só fico empacado mesmo e isso ta me encomodando demais. Eu pego o conteúdo, procuro, mas não consigo sair além de procurar o conteúdo de matemática discreta, AOC( Arquitetura e Organização de Computadores) estou empacado em coisas simples. Estou tendo um surto praticamente, rs.@Vitor-Felipe

**Comment by glauberramos** [comment]
Acho que tirar um pouco a pressão que tu tem sobre ti mesmo pode ajudar. Uma dica para aprender frontend é pensar que tu não ta estudando, mas pensar que é uma diversão, um passa tempo. 

Começa fazendo alguma coisa boba, tipo uma página sobre você mesmo, tenta criar algum personagem que você gosta em CSS, tenta fazer algum site para alguma coisa que você precisa (loja para vender coisas suas, suas fotos que tu tira, sei la, seja criativo :D).

Outra dica é reservar um horário para fazer isso, sei la tipo 2 horas por dia.

**Comment by fredmosc** [comment]
Bom, já passei por esse período de procrastinação intensa.. foi bem complicado, além da procrastinação a falta de foco estava me consumindo..  foi um periodo dificil em que outras coisas na vida pessoal estavam atrapalhando. Acabei procurando ajuda médica e foi de grande utilidade.

Além disso, instalei o plugin Wakatime que conta quanto tempo você programou no dia, funciona em diversas IDEs e é gratuito. Também instalei o Rescuetime que mede quanto tempo vc passa em sites e é categorizado se é produtivo ou não.

Em 1 ano usando esses dois plugins, passei de 9hrs de programação por semana para mais de 40hrs por semana e de 45% produtivo para 78% de produtividade. Ainda existe o StayFocusd que te bloqueia depois de determinado tempo em um site (esse não me adaptei muito)

E outra coisa q passei a fazer foi me programar semanalmente o q quero fazer, isso me ajudou demais também

**Comment by kvnol** [comment]
Puts, maravilhoso, @fredmosc. Vou começar a usar essas ferramentas, também.

**Comment by laurindo** [comment]
Galera, segue minha dica pra ajudar um pouco.
Acho que uma das melhores maneiras se somar estudo + diversão + aprender
muito é fazendo algum projeto. Por exemplo.

Tenta imaginar alguma aplicação web que vc gosta de usar. Pega a interface,
como a ferramenta lhe parece fácil de usar e tenta fazer. No início irá
surgir uma pancada de dúvidas, mas vc vai resolvendo de forma mais direta,
sem achar que o que vc tá aprendendo possa servir ou não para seu
currículo.
Outro benefício é que serve de portfólio ou de marco pra saber o quanto vc
evoluiu.
Depois é tirar um tempo pra estudar design pattern, estrutura de dados, e
teoria do porque vc está fazendo daquela maneira, etc.

Vlw.

**Comment by gabrieldarezzo** [comment]
O @grvcoelho fez uma talk bem bacana sobre procrastinação.

Veja se ajuda:
https://www.youtube.com/watch?v=tMuU6fViaS8&t=178s

-------

Sobre melhorar em programação uma coisa que deu muito certo comigo no inicio foi escolher uma tarefa grande, quebrar ela em diversas micro tarefas.

Vou exemplificar com um sistema de gerenciamento de cadeiras de cinema que exibe as cadeiras vagas.
 
Exiba 20 x 5 cadeiras 
 
Algo +/- assim:
 
![amcn3cxxg0yh1iiapjn8hlekeqavesqxwxhdhy7zdfa](https://user-images.githubusercontent.com/4562368/30278954-2e0bae60-96e2-11e7-8675-ce5661e8f70a.jpg)

 
Cada cadeira deve exibir o seu estado:
Verde = Livre
Vermelha = Ocupada
Amarela = Selecionada no momento
 
 
Para 'comprar' o ingresso precisa salvar um E-mail (Não pode repetir), senha (nada de texto em?!) e gerar um comprovante no termino do processo da compra, exibindo os acentos reservador.
  
Inicialmente pode fazer apenas com 1 sala,1 filme
 
Bônus:
Mudar opção de filmes, Salas, Horários

---

Tarefas menores:
- Iniciar algumas cadeiras já com estado.
- Ao clicar mudar o estado
- Enviar um E-mail simples
- Receber dados e printar na tela
- Receber dados e salvar 
- Receber dados comparar se é único e salvar 
....

Enfim acho que deu pra pegar o conceito dividir o 'problemão' em probleminhas.

O mais legal é a motivação que gera e você consegue enxergar a própria evolução do projeto, e claro praticar algo.

Espero que tenha ajudado haha.

Bons estudos e boa sorte.

**Comment by leticiacostadias** [comment]
Eu nunca comento nada aqui porque acho que não manjo do assunto, mas de procrastinação eu entendo bem.

Cara, eu sofro com isso constantemente também e, com o tempo, você aprende a lidar melhor com isso. Essa talk que indicaram é sensacional sobre o assunto. O fato é que a gente não lida bem com a ausência de deadline e estímulo positivo.
O que eu comecei a fazer pra me ajudar a procrastinar menos foi dividir as tarefas em pequenas metas, bem fracionado mesmo e me recompensar com coisas rápidas a cada tarefa ou meta, condicionando meu cérebro de que eu só poderia receber outra recompensa quando completasse outro objetivo.

No exemplo do @gabrieldarezzo você poderia se "premiar" ao terminar cada um dos passos com um vídeo curto no YouTube, um doce, uma volta no quarteirão, tocar uma música (se você toca algum instrumento), qualquer coisa que você usa hoje pra procrastinar por horas, mas numa dose pequena.

Aí entra a disciplina de conseguir voltar pra sua tarefa ou objetivo, condicionando o seu cérebro a esperar a recompensa e só consumir aquilo por pouco tempo somente quando você atinge um objetivo.

**Comment by rubenmarcus** [comment]
Tem um app chamado : **Forest** , uso no celular e no chrome, ele planta uma árvore (digital) a cada 120min que você fica sem usar o celular,ou navegar em redes sociais. Bom é um resultado legal você ver no final do dia uma floresta erguida.

O Asana é tão bom quanto o trello e ele te avisa dos projetos. Mas se você não ve muita motivação pra fazer código, vai aos poucos, tenta traduzir artigos e ajudar com open source, as vezes nos sentimos um peixe fora dágua, mas quando conseguir fazer algo vai se orgulhar e seguir em frente.

**Comment by ghost** [comment]
Poxa galera, valeu pelas, vou utilizar bastante. Inclusive, já estou indo atrás de todas as extensões! (: 
Pois bem, o @rubenmarcus  falou sobre projeto open source e é algo que tenho vontade em contribuir, tenho uma certa noção em ALP (Algoritmo e Lógica de Programação). Bem, estou cursando ADS e o professor tem passado durante a aula, Pascal. Sim, Pascal. Não sei o motivo mas, segundo ele, por ser mais "fácil" de ser aprendida. E minha questão fica em, como aprender uma linguagem nova? Para assim, eu poder contribuir, estudar algo mais "atual" que tem em diversas plataformas.  
Eu estou com bastante matéria acumulada na faculdade pelo fato de não ter conseguido me organizar direito desde o início. Estou pensando em utilizar a semana para estudos da faculdade, e final de semana para estudos fora, como uma "diversão" e passa-tempo. Tenho bastante interesse na área de segurança e é algo que quero estudar por fora. Vou utilizar a dica de todos (:
Estou motivado para sair desse estado de procrastinação, só falta sair! (: hahaha

**Comment by suissa** [comment]
@brunojokubauskas posso te dar uma dica que SEMPRE falo para meus alunos??

Pegue um pré-conhecimento que você tenha e aprenda como passar ele para programação, por exemplo:

Como vc entrou na faculdade agora com certeza você lembra de várias conteúdos de Matemática do Ensino Médio, aprenda a traduzir os conceitos e fórmulas que você conhece para JS, usando Node.js, por exemplo criar uma função de validação se um número é primo, com isso vc irá exercitar sua lógica matemática E booleana, isso eu te garanto.

Eu acredito que para você aprender realmente programar vc precisa concretizar os conhecimentos abstratos que vc já possui.

Outro exemplo seria você fazer uma função para calcular a área de um triângulo e reusando ela criar funções para calcular a área de outros polígonos, pois todos podem ser reduzidos a triângulos, menos a circunferência (apesar que essa tb pode só depende da infinidade de triângulos que vc usará ehheheheh)

Depois que vc aprender a criar funções vc pode ir para parte de estrutura de dados.

E em tudo que vc fizer vc precisará de lógica e algoritmos, mas isso vc pode aprender com o empirismo em vez de apenas aprender teoricamente, mas essa é apenas minha opinião de quem dá aula desde 2010.

Posso lhe indicar um roteiro:

Crie uma função para validar se é PAR usando negação do resto da divisão.
Crie uma função para validar se é ÍMPAR usando a função de PAR.
Crie uma função de potência sem usar Math.pow.
Crie uma função de radiciação usando apenas a função de potência, criada anteriormente.
Crie uma função para validar se é PRIMO usando as funções anteriores e as específicas para isso.
Crie uma função para resolver o Teorema de Pitágoras.
Cria uma função para resolver Bhaskara.

Isso acredito que já pode lhe dar um norte, caso vc ache a Matemática importante.

Depois de vc dominar a criação e utilização de funções vc pode partir para estrutura de dados.

Vc pode iniciar com o conceito de Conjuntos.

A partir disso vc já pode começar a usar map para fazer transformações entre conjuntos, depois podes usar o filter para filtrar valores de um Conjunto.

Por exemplo: crie um Array de números Naturais do 0 ao 100, depois use suas funções de par, ímpar e primo para filtrar os valores e criar Conjuntos novos.

E assim por diante.

**Comment by gabrielmanara** [comment]
@brunojokubauskas  no meu curso de Sistemas de Informação tive um professor de algoritmo sensacional. A linguagem que ele fazia a gente usar era C, toda aula ele dava um conceito novo e deixava um monte de exercícios pra gente resolver. Com o tempo ele foi preparando a gente para participar de maratonas de programação (nunca consegui ir pra uma regional, mas eu tentava), uma ferramenta muito boa se você quiser treinar e desenvolver sua parte de algoritmos é essa aqui http://br.spoj.com/ (Tem outras várias online, o legal é o desafio)

No geral o que eu te digo é **vá com calma!**, passei por esses mesmo problemas que você falou, procrastinação, de querer aprender vários conceitos e não conseguir. Foque na lógica, ela é mais importante do que qualquer linguagem, após isso comece a "brincar", pegue um shot no https://dribbble.com e tente fazer, pode ser CSS puro mesmo, faça no https://codepen.io/, você pode até compartilhar com a galera pedindo opinião do que o pessoal acha (cuidado, não desanime com os feedbacks negativos) depois com o tempo você pode começar a brinca com Stylus ou Sass. Faça um calculadora usando javascript, depois você pode começar a ver padrões de projetos, frameworks, es6, es7 etc, sempre se desafie. 

Mas cara **vá com calma!**! O processo é meio lento mesmo, mas com o tempo você vai começar a sentir prazer em fazer esses shots do dribble e brincar com javascript, dai o aprendizado vai fluindo. Se for algo muito forçado o conhecimento não flui e você começa a procrastinar. Com o tempo você vai começar a olhar para as coisas que você está fazendo e dizer "caralho ficou da hora" ou "caralho ficou top", aos poucos você vai conseguir absorver e consolidar seu conhecimento.

A partir de uma issue aqui no frontend br nasceu esse cara aqui https://github.com/training-center/mentoria, também sugiro você dar uma olhada. Mas como eu falei anteriormente, foque na lógica antes de mais nada.

**Comment by mathvbarone** [comment]
Cara, eu fiz um texto que pode te ajudar, não digo que vá curar a sua ansiedade, mas pelo menos é um norte pra você seguir e assim ir galgando os degraus rumo ao conhecimento.

https://medium.com/@matheusvbarone/e-se-hoje-fosse-meu-primeiro-dia-como-desenvolvedor-front-end-5fc5b19e2228

Espero que você goste :)

**Comment by synga** [comment]
Cara então o seu problema é 0 em tecnologia e 100% em você. Não importa quantos links e projetos e coisas legais a galera te passar pra estudar e fazer, você não vai.

Eu era exatamente como você cara, meus 4 anos e meio de fac00l (1 semestre a mais pois enrolei no TCC) foram no estilo "estudando pra passar" por ter preguiça de fazer as coisas, o trampo me cansava e eu só queria chegar e dormir por que tinha que acordar cedo no outro dia. Fui pegar gosto pela coisa quando arranjei um estagio e me apresentaram melhor as tecnologias e cobravam meu desempenho. Não desejo pra ninguém o que aconteceu comigo, me via desesperado, tento que aprender as pressas e sozinho, não sabia onde correr e quem consultar, quase desisti da área e tinha caído em publicidade pra vc ter uma ideia. Hoje estou a quase 1 ano trampando em front e cara amo demais, fora que estudo sempre, tenho meus projetos e consigo conciliar tudo por que aprendi meu ritmo. 

**Vou te sugerir umas coisas**

- Não fique em casa, não tente fazer as coisas em casa, pois em casa não sai e não da certo, quando você vê passou 50 minutos pela interwebs e não codou nada, fora que é confortável demais. Vai pra uma biblioteca, algum espaço publico onde você fique a vontade, mas não o suficiente pra ficar esquecer do que foi lá pra fazer.
- Cria uma planilha, tarefas, roadmaps ou o que seja sobre o que vc quer aprende ou o que quer seguir, planeje os estudos, veja do que cada coisa depende para você aprender. EX: Quer aprender Angular? Então você vai precisar aprender TypeScript, SASS, package manager, etc. Se quiser algo pra seguir eu gosto muito desse [roadmap do FreeCodeCamp](https://medium.freecodecamp.org/a-roadmap-to-becoming-a-web-developer-in-2017-b6ac3dddd0cf). Sigo ele e vou riscando as coisas que vou aprendendo a ponto de me sentir confiante pra usar em um trampo.
- Use sites bem didáticos pra aprender, de preferencia nada de cursos que você tem que ficar vendo videos e criando projetos na sua maquina, use um CodeCademy, FreeCodeCamp, CodeSchool. Tem alguns em inglês, se você manja vai ser mais fácil de achar bons conteúdos.
- **NÃO SE DESESPERE**, isso é um erro e só te atrasa, você ta começando a faculdade, vai seguindo, conhecendo, lendo, não precisa terminar a faculdade com o emprego dos sonhos ganho 15k por mês e sabendo codar um androide humanoide que reconhece emoções humanas. Cada pessoa tem seu ritmo e estagio ta ai pra ensinar, amigos também, fóruns, grupos no FaceBook, Twitter e mais. Vai aprendendo, vai aplicando o que aprendeu e vai seguindo.
- Isso de "consigo entender a matéria mas em casa eu enrosco" é normal, se você tem a lógica está ótimo, o resto aprende com o tempo.
- Aproveite a galera do front, nunca vi alguém responder mal, ter preguiça de ajudar, se irritar e o que seja. Pede ajuda e vai atrás, não fica esperando alguém te chegar com a formula mágica de como ser um bom dev, isso não existe e essa área é extremamente mutável para se ter algo assim. Além disso ninguém vai te vir com uma injeção de animo pra te tirar da cama e fazer suas coisas, isso vai ter que sair de você.
- Não querendo ser o chato e nem colocando pressão ou te desanimar, mas se vc não tem problemas psicológicos como depressão, ansiedade ou qualquer outra coisa que te prenda e te faça não ter vontade de nada e isso é realmente só preguiça então cara uma hora você vai ser passado pra trás e vai se ferrar bonito na área. Infelizmente não é uma área que da pra você terminar a faculdade e ficar só com o que aprendeu lá, vai ter que correr atrás O RESTO DA VIDA. Então cara realmente se anime por que é uma área incrível com poderes imensos e que te da gosto de trabalhar e estudar sobre, a cada projeto é algo bom que você sente, você só precisa conhecer isso. E caso ache que seu problema pode ser sim algo psicológico, vá atrás, vá descobrir o que é e pedir ajuda a um profissional, sem medo e vergonha disso.

Espero que te ajude, te esclareça em alguma coisa e boa sorte.

**Comment by leandroruel** [comment]
tá procrastinando com o que?

**Comment by rubenmarcus** [comment]
@synga

> 
Use sites bem didáticos pra aprender, de preferencia nada de cursos que você tem que ficar vendo videos e criando projetos na sua maquina, use um CodeCademy, FreeCodeCamp, CodeSchool. Tem alguns em inglês, se você manja vai ser mais fácil de achar bons conteúdos.


Po comprei 23 cursos de programação e design no Udemy e são todos em videos e nao me arrependo. Você tem alguma objeção tipo , não funciona pra voce? eu acho meio ruim pq as vezes parece vago, mas é bom q da pra vc rever a lição varias vezes e até perguntar pro dono do vídeo. EAD é assim em video a maioria, e funciona.

**Comment by calaca** [comment]
Concordo com o @rubenmarcus. Não vejo problema algum em vídeo-aulas. Agora, se o aluno ficar só assistindo aulas e não colocar a mão na massa, aí sim temos um problema.

**Comment by phenriqueleao** [comment]
Estou dando uma pesquisada sobre design então me desculpem reviver a issue.

Preciso defender o ponto de vídeo aulas também. Na época pré-YouTube, onde tínhamos praticamente só conteúdo em texto, eu tinha uma dificuldade imensa de aprendizado, pois a forma que eu mais consigo aprender é justamente com alguém me mostrando primeiro, e essa forma fica perfeita em vídeo.

**Comment by synga** [comment]
@rubenmarcus Cara esqueci total de colocar sobre vídeo-aulas hahahaha mas acho extremamente ótimo. Comprei uns na Udemy também, atualmente to fazendo um de Vue que é muito bom, super didático e aprende muito bem. Nada contra, só esqueci de mencionar mesmo.

Alias pra quem interessar tem um site bem legal que te passa exercícios randômicos para fazer. Te da o que você precisa fazer, pra quem e pra ajudar que tipos de pessoas:
https://designercize.com/
